src/main/java/com/github/fge/jsonpatch/ExtendedJsonPatch.java

```diff
@@ -0,0 +1,66 @@
+package com.github.fge.jsonpatch;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.ObjectReader;
+import com.fasterxml.jackson.databind.ObjectWriter;
+import com.fasterxml.jackson.databind.jsontype.NamedType;
+import com.github.fge.jackson.JacksonUtils;
+import com.github.fge.jsonpatch.operation.*;
+import com.github.fge.msgsimple.bundle.MessageBundle;
+import com.github.fge.msgsimple.load.MessageBundles;
+
+import java.io.IOException;
+
+public class ExtendedJsonPatch
```

Comment:
We're really close.  I hate having the reader/writer hanging off the top level interfaces in the program.  Basically last thing I'd do is move ExtendedJsonPatch to ExtendedJsonPatchFactory, take the statics off so you can pass around the fromJson in tests.  Add your StandardJsonPatchFactory back with the mapper code from JsonPatch with its from json.  Then deprecate the static fromJson on JsonPatch and and add a static StandardJsonPatchFactory to it.  (circular dependency but oh well).  Then let's just fucking land it.  I stared at the jackson docs all night.